File: SYP.Web/Modules/ServerTypes/Warehouse/WarehousesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WarehousesRow } from "./WarehousesRow";

export interface WarehousesColumns {
    Id: Column<WarehousesRow>;
    Code: Column<WarehousesRow>;
    Name: Column<WarehousesRow>;
    City: Column<WarehousesRow>;
    Phone: Column<WarehousesRow>;
    ManagerName: Column<WarehousesRow>;
    IsActive: Column<WarehousesRow>;
}

export class WarehousesColumns extends ColumnsBase<WarehousesRow> {
    static readonly columnsKey = 'Warehouse.Warehouses';
    static readonly Fields = fieldsProxy<WarehousesColumns>();
}

File: SYP.Web/Modules/Warehouse/Warehouses/WarehousesColumns.cs
namespace SYP.Warehouse.Columns;

[ColumnsScript("Warehouse.Warehouses")]
[BasedOnRow(typeof(WarehousesRow), CheckNames = true)]
public class WarehousesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public int Id { get; set; }

    [EditLink, Width(120)]
    public string Code { get; set; }

    [Width(200)]
    public string Name { get; set; }

    [Width(150)]
    public string City { get; set; }

    [Width(120)]
    public string Phone { get; set; }

    [Width(150)]
    public string ManagerName { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}
